File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/PartitionDependencyHash.java
package io.qimia.uhrwerk.common.model;

import net.openhft.hashing.LongHashFunction;

public class PartitionDependencyHash {

    /**
     * Generate a unique Id for a partition combination (assumes keys have been set)
     * @param childId the partition that has dependencies
     * @param parentId the partition the child is depending on
     * @return long value used as id / key
     */
    public static Long generateId(Long childId, Long parentId) {
        StringBuilder res = new StringBuilder().append(childId).append(parentId);
        return LongHashFunction.xx().hashChars(res);
    }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/model/PartitionDependencyHash.kt
package io.qimia.uhrwerk.common.model

import net.openhft.hashing.LongHashFunction

object PartitionDependencyHash {
    /**
     * Generate a unique Id for a partition combination (assumes keys have been set)
     * @param childId the partition that has dependencies
     * @param parentId the partition the child is depending on
     * @return long value used as id / key
     */
    fun generateId(childId: Long?, parentId: Long?): Long {
        val res = StringBuilder().append(childId).append(parentId)
        return LongHashFunction.xx().hashChars(res)
    }
}